roofit/roofitcore/inc/RooAbsReal.h

```diff
@@ -463,7 +463,35 @@ class RooAbsReal : public RooAbsArg {
 
   //--------------------------------------------------------------------
 
- protected:
+  // Experimental AD feature functions
+  protected:
+  /// the variable that stores the result expression for the object.
+  std::string _adResult = ""; ///<!
```

Comment:
We gotta find a way to do the code generation without this additional data member for `RooAbsReal`.

Cutting-edge fits have computation graph with thousands of `RooAbsReals`, and we should not increase their memory footprint without a good reason. And I think in this case, we can get around this. Especially because this is dead weight if AD is not used.

What comes to my mind is passing around a map of results in the `translate` function instead. But probably we will change the signature of `translate` more often for similar reasons then.

So what about passing around some `CodeGenContext` in `translate`? This could include the `globalScope` object that you already have, and also the map of results. The keys of the map can be strings with the `RooAbsArg::GetName()`, because in the likelihood created by the BatchModeHelpers you won't have duplicate names.